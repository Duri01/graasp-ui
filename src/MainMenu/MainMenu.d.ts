import React from 'react'
import { MenuItemType } from './MenuItem'

interface MainMenuProps<T> {
  id?: string
  children?: React.ReactElement
}

declare const MainMenu: React.FC<MainMenuProps>
