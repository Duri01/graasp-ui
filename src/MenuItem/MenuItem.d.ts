import React from 'react'

export interface MenuItemType {
  id: string
  text?: string
  icon?: React.ReactElement
  path?: string
  key?: string
  onClick?: (path?: string) => void
}

declare const MenuItem: React.FC<MenuItemType>
