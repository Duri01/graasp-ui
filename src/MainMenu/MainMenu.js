import React from 'react'
import List from '@material-ui/core/List'

export const MainMenu = ({ id, children }) => {
  return <List id={id}>{children}</List>
}

export default MainMenu
