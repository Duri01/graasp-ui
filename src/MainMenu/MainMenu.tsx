import React, { FC } from 'react';
import List from '@material-ui/core/List';

export interface MainMenuProps {
  id?: string;
  children?: React.ReactElement;
}

export const MainMenu: FC<MainMenuProps> = ({ id, children }) => {
  return <List id={id}>{children}</List>;
};

export default MainMenu;
