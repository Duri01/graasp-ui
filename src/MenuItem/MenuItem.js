import React from 'react';
import MaterialMenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

export const MenuItem = ({ id, text, icon, onClick }) => {
  return (
    <MaterialMenuItem id={id} onClick={onClick} button>
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      {text && <ListItemText primary={text} />}
    </MaterialMenuItem>
  );
};

export default MenuItem;
