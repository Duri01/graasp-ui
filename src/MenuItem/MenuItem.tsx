import React, { FC, SyntheticEvent, ReactElement } from 'react';
import MaterialMenuItem from '@material-ui/core/MenuItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';

export interface MenuItemProps {
  id?: string;
  text?: string;
  icon?: ReactElement;
  path?: string;
  key?: string;
  onClick?: (event?: SyntheticEvent) => void;
}

export const MenuItem: FC<MenuItemProps> = ({ id, text, icon, onClick }) => {
  return (
    <MaterialMenuItem id={id} onClick={onClick} button>
      {icon && <ListItemIcon>{icon}</ListItemIcon>}
      {text && <ListItemText primary={text} />}
    </MaterialMenuItem>
  );
};

export default MenuItem;
