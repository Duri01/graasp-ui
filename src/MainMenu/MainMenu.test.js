import React from 'react'
import { shallow } from 'enzyme'
import { MainMenu } from './MainMenu'
import MenuItem from '../MenuItem'
import List from '@material-ui/core/List'

const createMainMenuProps = ({ id } = {}) => ({
  id
})

const menuWithMenuItems = (
  <>
    <MenuItem key='one' text='one' />,
    <MenuItem key='two' text='two' />,
    <MenuItem key='three' text='three' />,
  </>
)

const menuWithCustomMenuItems = (
  <>
    <div key='one' text='one' />,
    <div key='two' text='two' />,
    <div key='three' text='three' />,
  </>
)

describe('MainMenu', () => {
  let wrapper

  describe('default values', () => {
    beforeAll(() => {
      const props = createMainMenuProps()
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MainMenu {...props} />)
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one <List /> with default values', () => {
      const list = wrapper.find(List)
      expect(list).toHaveLength(1)
    })
  })

  describe('menu is defined with graasp MenuItem', () => {
    beforeAll(() => {
      const props = createMainMenuProps({ id: 'id' })
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(<MainMenu {...props}>{menuWithMenuItems}</MainMenu>)
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one <List /> component with graasp menu', () => {
      const list = wrapper.find(List)
      expect(list).toHaveLength(1)
      expect(list.prop('id')).toEqual('id')
      const items = list.find(MenuItem)
      expect(items).toHaveLength(3)
    })
  })

  describe('menu is defined with custom component', () => {
    beforeAll(() => {
      const props = createMainMenuProps()
      // eslint-disable-next-line react/jsx-props-no-spreading
      wrapper = shallow(
        <MainMenu {...props}>{menuWithCustomMenuItems}</MainMenu>
      )
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one <List /> component with custom menu', () => {
      const list = wrapper.find(List)
      expect(list).toHaveLength(1)
      const items = list.find('div')
      expect(items).toHaveLength(3)
    })
  })
})
