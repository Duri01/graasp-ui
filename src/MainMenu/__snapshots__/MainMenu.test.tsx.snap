// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu default values renders correctly 1`] = `<WithStyles(ForwardRef(List)) />`;

exports[`MainMenu menu is defined with custom component renders correctly 1`] = `
<WithStyles(ForwardRef(List))>
  <div
    data-text="one"
    key="one"
  />
  ,
  <div
    data-text="two"
    key="two"
  />
  ,
  <div
    data-text="three"
    key="three"
  />
  ,
</WithStyles(ForwardRef(List))>
`;

exports[`MainMenu menu is defined with graasp MenuItem renders correctly 1`] = `
<WithStyles(ForwardRef(List))
  id="id"
>
  <MenuItem
    key="one"
    text="one"
  />
  ,
  <MenuItem
    key="two"
    text="two"
  />
  ,
  <MenuItem
    key="three"
    text="three"
  />
  ,
</WithStyles(ForwardRef(List))>
`;
